source/module_esolver/esolver_ks_lcao.cpp

```diff
@@ -555,6 +557,12 @@ namespace ModuleESolver
         // update real space Hamiltonian
         this->p_hamilt->refresh();
     }
+
+    if (GlobalV::sc_mag_switch && iter > 1)
```

Comment:
Would you like to add some annotations in esolver for developers to read?

Reply:
done